drivers/at86rf2xx/at86rf2xx_netdev.c

```diff
@@ -130,59 +148,115 @@ static int _send(netdev_t *netdev, const iolist_t *iolist)
+     * */
     if (buf == NULL) {
+        /* TODO Is this not a problem? Frame Buffer release */
         at86rf2xx_fb_stop(dev);
+        DEBUG("Packet lenght: %d\n", pkt_len);
         return pkt_len;
     }
     /* not enough space in buf */
     if (pkt_len > len) {
         at86rf2xx_fb_stop(dev);
+#ifdef MODULE_AT86RFR2
+        /* If Upper layer does not supply a big enough buffer data loss is
+         * unavoidable clear frame buffer protection
+         * */
+        *AT86RF2XX_REG__TRX_CTRL_2 &= ~(1 << RX_SAFE_MODE);
+#endif
         return -ENOBUFS;
     }
     #ifdef MODULE_NETSTATS_L2
-        netdev->stats.rx_count++;
-        netdev->stats.rx_bytes += pkt_len;
+    netdev->stats.rx_count++;
+    netdev->stats.rx_bytes += pkt_len;
     #endif
     /* copy payload */
     at86rf2xx_fb_read(dev, (uint8_t *)buf, pkt_len);
 
+#ifndef MODULE_AT86RFR2
     /* Ignore FCS but advance fb read - we must give a temporary buffer here,
      * as we are not allowed to issue SPI transfers without any buffer */
     uint8_t tmp[2];
     at86rf2xx_fb_read(dev, tmp, 2);
     (void)tmp;
+#endif
 
+    /* at86rf212  RSSI_BASE_VAL +3.1*RSSI, base varies for diff. modulation and datarates
+     * at86RF232  RSSI_BASE_VAL +3.0*RSSI, base -91dBm
+     * at86RF233  RSSI_BASE_VAL +3.0*RSSI, base -94dBm
+     * at86RF231  RSSI_BASE_VAL +3.0*(RSSI-1), base -91dBm
+     * at***RFR2  RSSI_BASE_VAL +3.0*(RSSI-1), base -90dBm
+     *
+     * AT86RF233 MAN. p.99  It is not recommended reading the RSSI value when using
+     * the Extended Operating Modes, use ED instead
+     * at86RF231  RSSI_BASE_VAL +ED, base -90dBm
+     * at86RF232  RSSI_BASE_VAL +ED, base -91dBm
+     * at86RF233  RSSI_BASE_VAL +ED, base -94dBm
+     * at***RFR2  RSSI_BASE_VAL +ED, base -90dBm
+     *
+     * Ensuring timing for ED read out is critical. But neglected here.
+     * ATmega256/128/64RFR2 manual p.75 The ED value needs to be read within 224 μs
+     * including the ED measurement time after the TRX24_RX_END interrupt.
+     */
     if (info != NULL) {
         uint8_t rssi = 0;
         netdev_ieee802154_rx_info_t *radio_info = info;
+
+#if defined(MODULE_AT86RFR2)
+        rssi = PHY_ED_LEVEL;
+        /* skip Payload + FCS */
+        at86rf2xx_sram_read(dev, pkt_len + 2, &(radio_info->lqi), 1);
+        /* clear frame buffer protection */
+        at86rf2xx_fb_stop(dev);
+#elif defined(MODULE_AT86RF231) || defined(MODULE_AT86RF232) || defined(MODULE_AT86RF233)
         at86rf2xx_fb_read(dev, &(radio_info->lqi), 1);
-#ifndef MODULE_AT86RF231
-        at86rf2xx_fb_read(dev, &(rssi), 1);
         at86rf2xx_fb_stop(dev);
+        rssi = at86rf2xx_reg_read(dev, AT86RF2XX_REG__PHY_ED_LEVEL);
 #else
+        at86rf2xx_fb_read(dev, &(radio_info->lqi), 1);
+        at86rf2xx_fb_read(dev, &(rssi), 1);
         at86rf2xx_fb_stop(dev);
-        rssi = at86rf2xx_reg_read(dev, AT86RF2XX_REG__PHY_ED_LEVEL);
+        rssi = 3 * rssi;
```

Comment:
The changes here and above after `#elif` I think are out of scope, if I'm not wrong, they "fix" the behaviour on at86rf2xx devices, isn't it?

Reply:
Yeah, that's nice, but it should come in a follow up PR, since this one is about adding support for the at86rfr2 driver.

Reply:
I will prepare this, can we merge that beforehand, so that a can adjust the at86rfr2 driver rebased on that?

Reply:
Yes, if you want to clean up at86rf2xx code first it would be much easier to review this PR.